torch/_higher_order_ops/foreach_map.py

```diff
@@ -0,0 +1,22 @@
+# mypy: allow-untyped-decorators
+# mypy: allow-untyped-defs
+from torch._higher_order_ops.prim_hop_base import FunctionWithNoFreeVars, PrimHOPBase
+
+
+class ForeachMap(PrimHOPBase):
+    def __init__(self):
+        super().__init__("foreach_map")
+
+    def __call__(self, fn, operands, *unused, **kwargs):  # type: ignore[override]
+        fn = FunctionWithNoFreeVars(fn)
+        return super().__call__(fn, operands, **kwargs)
+
+
+_foreach_map = ForeachMap()
+
+
+def foreach_map(op, operands, *unused, **kwargs):
```

Comment:
docstring, type annotations?

Reply:
do we want the first input here to be op ? Eventually we'd like this to support more arbitrary functions right ?

Reply:
I can change the arg name, it actually supports arbitrary functions today.

Reply:
The operands can kind of be anything so I didn't provide type annotations for this, I will add a docstring though.

Reply:
Callable, Any, Any i guess.